tests/unittest_observer/FaceliftObserverTest.cpp

```diff
@@ -0,0 +1,76 @@
+#include <gtest/gtest.h>
+#include "IPCProxyBase.h"
+#include "InterfaceBase.h"
+#include <QSignalSpy>
+
+namespace {
+
+using namespace facelift;
+
+class Counter
+{
+public:
+    Counter() { m_value = 0; }
+
+    void incValue() {
+        m_value++;
+    }
+    int getValue() const {
+       return m_value;
+    }
+private:
+    int m_value;
+};
+
+class IPCProxyBaseTest : public ::testing::Test
+{
+public:
+    Counter c1;
+    Counter c2;
+    StandartObserver  < std::function<void()> > * obs1 = new StandartObserver  < std::function<void()> > (std::bind(&Counter::incValue, &c1) );
+    SingleTimeObserver< std::function<void()> > * obs2 = new SingleTimeObserver< std::function<void()> > (std::bind(&Counter::incValue, &c2) );
+
+    IPCProxyBase<InterfaceBase> proxyBase{nullptr};
+
+    ~IPCProxyBaseTest() {
+        delete obs1;
```

Comment:
Do you have some specific reason for manual memory management ?

Reply:
This is just for test example.

Reply:
Do you mean that it is not final edition or it is doesn't matter what to write in test examples ?

Reply:
I suppose that writing in test cases should be as simple as possible with obviously stages.

Reply:
not always!
I agree with some examples like  that 
`   int addition(int a, int b);
    int x1 = addition(1,2);
    int x2 = addition(2,3);

     EXPECT_EQ(3, x1);
     EXPECT_EQ(5, x2);
`
where it is in one small place, but when it is inside some class, and may result to memory leaks, exposed in the public repository, I don't agree that test can be justification
Anyway it looks messy. 
And what if someone decides to add
another pointer and forget to delete it, or
will add an array of pointers?

And btw
using smart pointers, auto keyword, range-based loops etc is easier
both - for reading and writing.